src/components/Drawer/Drawer.stories.tsx

```diff
@@ -0,0 +1,311 @@
+
+export default {
+	title: "Components/Drawer",
+	component: Drawer,
+} as Meta<DrawerProps>;
+
+export const Default = () => {
+	const [defaultDrawerOpen, setDefaultDrawerOpen] = useState(false);
+	const [noDismissDrawerOpen, setNoDismissDrawerOpen] = useState(false);
+	const [hasBackButtonDrawerOpen, sethasBackButtonDrawerOpen] = useState(false);
+
+	const toggleDrawerByName = (drawerName: string) => {
+		switch (drawerName) {
+			case "default":
+				setDefaultDrawerOpen(!defaultDrawerOpen);
+				break;
+			case "no-dismiss":
+				setNoDismissDrawerOpen(!noDismissDrawerOpen);
+				break;
+			case "back-button":
+				sethasBackButtonDrawerOpen(!hasBackButtonDrawerOpen);
+				break;
+		}
+	};
+
+	return (
+		<main>
+			<section
+				style={{
+					display: "flex",
+					justifyContent: "space-between",
+					marginBottom: "1rem",
+				}}
+			>
+				<Button onClick={() => toggleDrawerByName("default")}>
+					Open Default Drawer
+				</Button>
+
+				<Button onClick={() => toggleDrawerByName("no-dismiss")}>
+					Open Drawer dismiss disabled
+				</Button>
+
+				<Button onClick={() => toggleDrawerByName("back-button")}>
+					Open Drawer with Back button
+				</Button>
+			</section>
+			<section>
+				<TextArea
+					helperText="Try typing here while scrim is on."
+					label="Test focus by typing here"
+					maxLength={10}
+					placeholder="Placeholder"
+					translations={{
+						over: "over",
+						remaining: "remaining",
+					}}
+				/>
+			</section>
+
+			<Drawer
+				open={defaultDrawerOpen}
+				onClose={() => toggleDrawerByName("default")}
+				title="Drawer with default behavior"
+			>
+				<p>This Drawer should only have the x close button</p>
+			</Drawer>
+
+			<Drawer
+				open={noDismissDrawerOpen}
+				onClose={() => toggleDrawerByName("no-dismiss")}
+				title="Drawer with dimiss on scrim click disabled"
+				closeOnBackgroundClick={false}
+			>
+				<p>This Drawer will not close if you click on the scrim background</p>
+			</Drawer>
+
+			<Drawer
+				open={hasBackButtonDrawerOpen}
+				onClose={() => toggleDrawerByName("back-button")}
+				onBack={() => {
+					alert("Back button pressed");
+				}}
+				title="With back button"
+			>
+				<p>This Drawer both a Back button and the x close button</p>
+			</Drawer>
+		</main>
+	);
+};
+
+export const withForm = () => {
+	const [formDrawerOpen, setFormDrawerOpen] = useState(false);
+
+	return (
+		<main>
+			<section
+				style={{
+					display: "flex",
+					justifyContent: "space-between",
+					marginBottom: "1rem",
+				}}
+			>
+				<Button onClick={() => setFormDrawerOpen(!formDrawerOpen)}>
+					Toggle Drawer Open
+				</Button>
+			</section>
+
+			<Drawer
+				open={formDrawerOpen}
+				onClose={() => setFormDrawerOpen(false)}
+				title="Submission Form"
+				closeOnBackgroundClick={false}
+			>
+				<Form
+					// biome-ignore lint/suspicious/noExplicitAny: we require maximum flexibility here
+					onSubmit={(e: any) => {
```

Comment:
```suggestion
					// biome-ignore lint/suspicious/noExplicitAny: we require maximum flexibility here
					onSubmit={(e: FormEvent<HTMLFormElement>) => {
```

Per a quick "find all" for `onsubmit` ☝️

Reply:
fixed